explorer/src/components/ClusterModal.tsx

```diff
@@ -66,36 +66,66 @@ function CustomClusterInput({ activeSuffix, active }: InputProps) {
     };
   };
 
-  const onUrlInput = useDebounceCallback((url: string) => {
-    updateCustomUrl(url);
-    if (url.length > 0) {
-      query.set("cluster", "custom");
-      history.push({ ...location, search: query.toString() });
-    }
-  }, 500);
+  const onUrlInput = useDebounceCallback(
+    (url: string, solarweaveUrl: string) => {
+      updateCustomUrl(url, solarweaveUrl);
+      if (url.length > 0) {
+        query.set("cluster", "custom");
+        history.push({ ...location, search: query.toString() });
+      }
+    },
+    500
+  );
 
   const inputTextClass = editing ? "" : "text-muted";
   return (
-    <Link
-      to={(location) => clusterLocation(location)}
-      className="btn input-group input-group-merge p-0"
-    >
-      <input
-        type="text"
-        defaultValue={customUrl}
-        className={`form-control form-control-prepended ${inputTextClass} ${customClass(
-          "border"
-        )}`}
-        onFocus={() => setEditing(true)}
-        onBlur={() => setEditing(false)}
-        onInput={(e) => onUrlInput(e.currentTarget.value)}
-      />
-      <div className="input-group-prepend">
-        <div className={`input-group-text pr-0 ${customClass("border")}`}>
-          <span className={customClass("text") || ""}>Custom:</span>
+    <>
+      <Link
+        to={(location) => clusterLocation(location)}
+        className="btn input-group input-group-merge p-0"
+      >
+        <input
+          type="text"
+          placeholder="Solana RPC URL"
+          defaultValue={customUrl}
+          className={`form-control form-control-prepended ${inputTextClass} ${customClass(
+            "border"
+          )}`}
+          onFocus={() => setEditing(true)}
+          onBlur={() => setEditing(false)}
+          onInput={(e) => onUrlInput(e.currentTarget.value, solarweaveUrl)}
+        />
+
+        <div className="input-group-prepend">
+          <div className={`input-group-text pr-0 ${customClass("border")}`}>
+            <span className={customClass("text") || ""}>Custom:</span>
+          </div>
         </div>
-      </div>
-    </Link>
+      </Link>
+
+      <Link
+        to={(location) => clusterLocation(location)}
+        className="btn input-group input-group-merge p-0"
+      >
+        <input
+          type="text"
+          placeholder="Solarweave Database"
```

Comment:
Do you expect it to be common to use other databases? I was expecting there to be a single arweave database for each cluster.

Reply:
If someone is querying a custom RPC node.
It might also be a forked chain.

Thus, I think a custom Solarweave Database would make sense!